Makefile

```diff
@@ -35,7 +35,7 @@ mod:
 	${GO_CMD} go mod init github.com/${GIT_ORG}/${GIT_REPO}
 	${GO_CMD} go mod tidy
 
-test:
+test: mod
```

Comment:
As a first pass I've made `mod` a prerequisite for `test` and removed it from the prerequisites list for `generate` in the SDK Makefile(s).  This works for now but I think the actual race involves `mod` running before `remove-unused` has finished removing the generated `<service>/go.(mod|sum)` files.  A more direct fix might be to have the `mod` task remove generated service-specific `go.mod` and `go.sum` files instead.

Reply:
Just checked this out and tried building on my linux server, seems to run fine at least.

Reply:
I ended up updating the `mod` task to clean up the extra `go.mod` and `go.sum` files and reordering the prerequisites for `generate` and that approach seems to work to.  Let me know if you prefer the earlier version of this PR instead, though.

Reply:
No, I prefer this way.